ibis/backends/bigquery/compiler.py

```diff
@@ -690,3 +690,44 @@ def visit_DropColumns(self, op, *, parent, columns_to_drop):
         table = sg.to_identifier(parent.alias_or_name, quoted=quoted)
         column = sge.Column(this=star, table=table)
         return sg.select(column).from_(parent)
+
+    def visit_TableUnnest(
+        self, op, *, parent, column, offset: str | None, keep_empty: bool
+    ):
+        quoted = self.quoted
+
+        column_alias = sg.to_identifier(
+            util.gen_name("table_unnest_column"), quoted=quoted
+        )
+
+        selcols = []
+
+        table = sg.to_identifier(parent.alias_or_name, quoted=quoted)
+
+        opname = op.column.name
+        if opname in op.parent.schema:
```

Comment:
I don't fully understand the purpose of these different branches here or in the other implementations. Could you add a comment?

Is it related to avoiding name collisions between the original column being unnested and the unnested version of it?

Reply:
> Is it related to avoiding name collisions between the original column being unnested and the unnested version of it?

That's exactly it. I can add some commentary.

Reply:
Done